.github/workflows/update-caches.yml

```
@@ -0,0 +1,136 @@
+  NIGHTLY_TOOLCHAIN: nightly
+
+jobs:
+  env:
+    runs-on: ubuntu-latest
+    outputs:
+      NIGHTLY_TOOLCHAIN: ${{ steps.env.outputs.NIGHTLY_TOOLCHAIN }}
+      MSRV: ${{ steps.env.outputs.MSRV }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - name: get MSRV
+        id: msrv
+        run: |
+          msrv=`cargo metadata --no-deps --format-version 1 | jq --raw-output '.packages[] | select(.name=="bevy") | .rust_version'`
+          echo "MSRV=$msrv" >> $GITHUB_OUTPUT
+      - name: Expose Env
+        id: env
+        run: |
+          echo "NIGHTLY_TOOLCHAIN=${{ env.NIGHTLY_TOOLCHAIN }}" >> $GITHUB_OUTPUT
+
+  build-caches:
+    name: Build Caches
+    needs: ["env"]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            toolchain: stable
+            target: ""
+          - os: macos-latest
+            toolchain: stable
+            target: ""
+          - os: windows-latest
+            toolchain: stable
+            target: ""
+          - os: ubuntu-latest
+            toolchain: ${{ needs.env.outputs.NIGHTLY_TOOLCHAIN }}
+            target: ""
+          - os: ubuntu-latest
+            toolchain: ${{ needs.env.outputs.MSRV }}
+            target: ""
+          - os: macos-latest
+            toolchain: ${{ needs.env.outputs.NIGHTLY_TOOLCHAIN }}
+            target: ""
+          - os: ubuntu-latest
+            toolchain: ${{ needs.env.outputs.NIGHTLY_TOOLCHAIN }}
+            target: wasm32-unknown-unknown
+          - os: ubuntu-latest
+            toolchain: stable
+            target: wasm32-unknown-unknown
+          - os: ubuntu-latest
+            toolchain: stable
+            target: x86_64-unknown-none
+          - os: ubuntu-latest
+            toolchain: stable
+            target: thumbv6m-none-eabi
+          - os: ubuntu-latest
+            toolchain: stable
+            target: aarch64-linux-android
+          - os: macos-latest
+            toolchain: stable
+            target: aarch64-apple-ios-sim
+
+    steps:
+      - name: Get Date
+        id: get-date
+        run: |
+          echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: Checkout Bevy main branch
+        uses: actions/checkout@v4
+        with:
+          repository: "bevyengine/bevy"
+
+      - name: Setup Rust
+        id: rust
+        uses: dtolnay/rust-toolchain@main
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          target: ${{ matrix.target }}
+
+      - name: Create lock file
+        run: cargo update
+
+      - name: Install Bevy dependencies
+        if: runner.os == 'linux'
+        run: |
+          sudo apt-get update;
+          DEBIAN_FRONTEND=noninteractive sudo apt-get install --no-install-recommends -yq \
+            libasound2-dev libudev-dev libxkbcommon-x11-0;
```

```suggestion
        uses: ./.github/actions/install-linux-deps
        with:
          wayland: true
          xkb: true
```

To reduce redundancy, we should use our existing [`install-linux-deps`](https://github.com/bevyengine/bevy/blob/2bddbdfd7c920d1ea61245dcdb7ff1c155e6b03b/.github/actions/install-linux-deps/action.yml) action. I'm not sure whether you want `wayland` to be installed or not, though.